### PR TITLE
records: add all pid_types

### PR DESCRIPTION
**Pull request by monaawi**
* Added jobs pid_type, record_Class, item and search endpoints 

* INSPIR-2145

### CODE REVIEW

File: inspirehep/config.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def _(x):
 
 # Deal with inconcistency :puke:
 PID_TYPES_TO_ENDPOINTS = {"lit": "literature", "aut": "authors"}
```
Comment (drjova) [inline inspirehep/config.py]:
```suggestion
PID_TYPES_TO_ENDPOINTS = {"lit": "literature", "aut": "authors", "job": "jobs"}
```

File: inspirehep/config.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ def _(x):
...
-PID_TYPES_TO_SCHEMA = {"hep": "lit", "authors": "aut"}
+PID_TYPES_TO_SCHEMA = {"hep": "lit", "authors": "aut", "jobs": "job"}
```
Comment (drjova) [inline inspirehep/config.py]:
```suggestion
PID_TYPES_TO_SCHEMA = {"hep": "lit", "authors": "aut", "job": "jobs"}
```

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -105,7 +105,11 @@ def test_with_record(base_app, create_record)
 
     def _create_record(record_type, data=None, with_pid=True, with_indexing=False):
         # FIXME: find a better location
-        MAP_PID_TYPE_TO_INDEX = {"lit": "records-hep", "aut": "records-authors"}
+        MAP_PID_TYPE_TO_INDEX = {
```
Comment (drjova) [inline tests/integration/conftest.py]: It's time to move this in ``config.py``

File: inspirehep/records/api/jobs.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+from ...pidstore.api import PidStoreJobs
+from .base import InspireRecord
+
+
+class JobsRecord(InspireRecord):
+    """Authors Record."""
+
+    pid_type = "job"
+
+    es_serializer = "RecordMetadataSchemaJSONV1"
```
Comment (drjova) [inline inspirehep/records/api/jobs.py]: This is wrong, we should add ``None`` for the moment to be explicit
```suggestion
    es_serializer = None
```

**Comment by drjova** [comment]
@monaawi about the commit message, should have a bit more details about the minter, REST endpoints etc

File: inspirehep/config.py (hunk 2 above)
Comment (monaawi) [reply inspirehep/config.py]: Actually here the field should be renamed to SCHEMA_TO_PID_TYPES and it should remain "jobs": "job"

Comment (drjova) [reply inspirehep/config.py]: Indeed, it's only used here https://github.com/inspirehep/inspirehep/blob/2588356b81ffa8a92ebb8a65d4d54b15dcc42e9b/inspirehep/pidstore/api/base.py#L70 could you please also refactor this part? it's just a small renaming.

Comment (monaawi) [reply inspirehep/config.py]: yes, of course

**Pull request merged**